### DISCUSSION TITLE
Any way to disable dns resolution?

### DISCUSSION THREAD

**Discussion by sgpublic** [abraunegg/onedrive#2758] [Q&A] [answered]
Hello everyone, I am a user from China. As we all know, China's special network environment may cause difficulties in accessing OneDrive, so I use a proxy to solve these difficulties, including dealing with DNS pollution.

However, the OneDrive business version in my area is not affected by the network environment and can be accessed directly, while [abraunegg/onedrive](https://github.com/abraunegg/onedrive) seems to handle DNS resolution by itself. In many cases, it will resolve the domain name IP of OneDrive and then use the IP to access directly. These requests bypass the rules set in my proxy that require OneDrive to connect directly, thereby consuming the traffic of my proxy.

I want to know if I can disable the DNS resolution of [abraunegg/onedrive](https://github.com/abraunegg/onedrive) so that it always uses the domain name to access, so as to hit my proxy rules and always connect directly? Thank you.

**Comment by abraunegg** [comment]
@sgpublic 
Please review the documentation: 

* https://github.com/abraunegg/onedrive/blob/master/docs/USAGE.md#access-onedrive-service-through-a-proxy

**Reply by sgpublic** [reply]
> Please review the documentation:

Sorry I didn't explain it clearly, the method I use is to deploy the proxy on the router to achieve transparent proxy, so I should not need to use HTTP_PROXY to use the proxy. This is why I asked whether I can disable the built-in DNS resolution of the software.

**Reply by abraunegg** [reply]
Then your transparent proxy is not operating correctly. It should be intercepting the traffic in the correct manner based on the destination being requested

**Reply by sgpublic** [reply]
> Then your transparent proxy is not operating correctly.

My rules contain these, they exist from the beginning: ('Domestic' is set to 'DIRECT'):

```
rules:
# OneDrive E5
- DOMAIN-SUFFIX,xxx-my.sharepoint.com,Domestic
- SRC-IP-CIDR,13.107.136.10/32,Domestic # This line was added after the troubleshooting mentioned below
# OneDrive 21vianet
- DOMAIN-SUFFIX,xxx-my.sharepoint.cn,Domestic
...
```

I just restarted onedrive and it's working fine for now.

![QQ_1721701177407](https://github.com/user-attachments/assets/0b6b0b64-8f4d-4df5-ba82-d78e14352748)

But if it runs for a long time, it may still be converted to IP access, such as 13.107.136.10, 13.107.138.10, etc, I don’t know if there will be new IPs in the future. For example, this picture was taken when I was troubleshooting a problem some time ago (normally I would set 'Proxy' to a proxy node instead of 'DIRECT'):

![786f17029bec18eda9c528e1f0377da8](https://github.com/user-attachments/assets/debba843-b4d0-4ffa-a77a-2a5a766e9dfe)


In fact, my OneDrive E5 domain name is currently directly accessible in my area, and it should not go through a proxy...

**Reply by abraunegg** [reply]
> But if it runs for a long time, it may still be converted to IP access, such as 13.107.136.10, 13.107.138.10, etc, I don’t know if there will be new IPs in the future. 

So create an internal DNS Server that will respond DNS queries for your sharepoint site with the IP addresses you want to use / have the service resolve to.

The issue is Microsoft Graph API has a number of resolution points - so you will have to then constantly chase the changes Microsoft are making.

**Reply by sgpublic** [reply]
The proxy I use supports DNS resolution hijacking to prevent pollution. It only needs to set the DNS server of the device connected to the network to the router, so the DNS resolution should always get the correct IP address.

Could I confirm this? Will [abraunegg/onedrive](https://github.com/abraunegg/onedrive) resolve the DNS by itself and convert it to IP access? If not, I think I haven't found the real problem and I should check other places.

**Reply by abraunegg** [reply, marked as answer]
The client has *zero* internal DNS resolution handling.

The client 100% relies on libcurl performing DNS resolution , which will *use* and *resolve* addresses based on the DNS servers configured on the host operating system.

**Reply by sgpublic** [reply]
Oh, it seems I didn't find the real problem, thank you very much for your reply.

**Reply by abraunegg** [reply]
To ensure you are correctly resolving - and not using then IPv6 by mistake, configure the client to only use IPv4
